File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    for i in range(1,number+1):
        c = 0
        for j in range(1,int(i**0.5)+1):
            if (i % j == 0):
                c += 1
                if j != i//j:
                    c += 1
        if c > limit:
            answer += power
            print(power)
        else:
            answer += c

    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for (int i = 1; i <= number; i++) {
            int c = 0;
            for (int j = 1; j <= Math.sqrt(i); j++) { 
                if (i % j == 0) {
                    c += 1;
                    if (j != i / j) {
                        c += 1;
                    }
                }
            }
            if (c > limit) {
                answer += power;
            } else {
                answer += c;
            }
        }
        return answer;
    }
}
